qcodes/instrument_drivers/rigol/DG1062.py

```diff
@@ -0,0 +1,345 @@
+
+    def inner(**inner_kwargs):
+        ex(**inner_kwargs)
+
+    inner.__doc__ = docstring
+
+    return inner
+
+
+class DG1062Burst(InstrumentChannel):
+    """
+    Burst commands for the DG1062. We make a separate channel for these to
+    group burst commands together.
+    """
+
+    def __init__(self, parent: 'DG1062', name: str, channel: int) ->None:
+        super().__init__(parent, name)
+        self.channel = channel
+
+        self.add_parameter(
+            "on",
+            get_cmd=f":SOUR{channel}:BURS?",
+            set_cmd=f":SOUR{channel}:BURS {{}}",
+            vals=vals.Enum(0, 1, "ON", "OFF")
+        )
+
+        self.add_parameter(
+            "polarity",
+            get_cmd=f":SOUR{channel}:BURS:GATE:POL?",
+            set_cmd=f":SOUR{channel}:BURS:GATE:POL {{}}",
+            vals=vals.Enum("NOR", "INV")
+        )
+
+        self.add_parameter(
+            "period",
+            get_cmd=f":SOUR{channel}:BURS:INT:PER ?",
+            set_cmd=f":SOUR{channel}:BURS:INT:PER {{}}",
+            vals=vals.MultiType(
+                vals.Numbers(min_value=3E-6, max_value=500),
+                vals.Enum("MIN", "MAX")
+            )
+        )
+
+        self.add_parameter(
+            "mode",
+            get_cmd=f":SOUR{channel}:BURS:MODE?",
+            set_cmd=f":SOUR{channel}:BURS:MODE {{}}",
+            vals=vals.Enum("TRIG","INF", "GAT")
+        )
+
+        self.add_parameter(
+            "ncycles",
+            get_cmd=f":SOUR{channel}:BURS:NCYC?",
+            set_cmd=f":SOUR{channel}:BURS:NCYC {{}}",
+            vals=vals.Numbers(min_value=1, max_value=500000)
+        )
+
+        self.add_parameter(
+            "phase",
+            get_cmd=f":SOUR{channel}:BURS:PHAS?",
+            set_cmd=f":SOUR{channel}:BURS:PHAS {{}}",
+            vals=vals.Numbers(min_value=0, max_value=360)
+        )
+
+        self.add_parameter(
+            "time_delay",
+            get_cmd=f":SOUR{channel}:BURS:TDEL?",
+            set_cmd=f":SOUR{channel}:BURS:TDEL {{}}",
+            vals=vals.Numbers(min_value=0)
+        )
+
+        self.add_parameter(
+            "trigger_slope",
+            get_cmd=f":SOUR{channel}:BURS:TRIG:SLOP?",
+            set_cmd=f":SOUR{channel}:BURS:TRIG:SLOP {{}}",
+            vals=vals.Enum("POS", "NEG")
+        )
+
+        self.add_parameter(
+            "source",
+            get_cmd=f":SOUR{channel}:BURS:TRIG:SOUR?",
+            set_cmd=f":SOUR{channel}:BURS:TRIG:SOUR {{}}",
+            vals=vals.Enum("INT", "EXT", "MAN")
+        )
+
+        self.add_parameter(
+            "idle",
+            get_cmd=f":SOUR{channel}:BURST:IDLE?",
+            set_cmd=f":SOUR{channel}:BURST:IDLE {{}}",
+            vals=vals.MultiType(
+                vals.Enum("FPT", "TOP", "BOTTOM", "CENTER"),
+                vals.Numbers()  # DIY
+            )
+        )
+
+    def trigger(self) ->None:
+        """
+        Send a soft trigger to the instrument. This only works if the trigger
```

Comment:
"soft" means "from software"/"from the driver"?

Reply:
Yes. I will make this more clear